workflows/energyplus.py

```diff
@@ -195,40 +197,67 @@ def run_energyplus(self, isIP, run_directory, file_name, args):
         if is_found:
-            if numeric_version >= 80300:  # EnergyPlus 8.3.0 was the first with command line options.
-
-                # start with the binary name, obviously
+            if by_api:
+                # if calling through API, then the arguments don't include the E+ binary
+                command_line_args = []
+            else:
                 command_line_args = [energyplus_binary]
 
-                if extension == '.imf':
-                    command_line_args += ['--epmacro']
+            if extension == '.imf':
+                command_line_args += ['--epmacro']
 
-                if extension != '.epJSON':
-                    command_line_args += ['--expandobjects']
+            # should be able to do this once we get pyExpandObjects going
+            if extension != '.epJSON':
+                command_line_args += ['--expandobjects']
 
-                # if not isIP:  # if using SI output units just use readvars CLI option
-                #     command_line_args += ['--readvars']
+            # TODO: add -d run_directory support to the E+ call here
 
-                # add some config parameters
-                command_line_args += ['--output-prefix', file_name_no_ext, '--output-suffix', 'C']
+            # if not isIP:  # if using SI output units just use readvars CLI option
+            #     command_line_args += ['--readvars']
 
-                # add in simulation control args
-                if 'weather' in args and args['weather']:
-                    command_line_args += ['--weather', args['weather']]
-                else:
-                    command_line_args += ['--design-day']
-
-                # and at the very end, add the file to run
-                command_line_args += [full_file_path]
+            # add some config parameters
+            command_line_args += ['--output-prefix', file_name_no_ext, '--output-suffix', 'C']
 
-                # run E+ and gather (for now fake) data
+            # add in simulation control args
+            if 'weather' in args and args['weather']:
+                command_line_args += ['--weather', args['weather']]
+            else:
+                command_line_args += ['--design-day']
+
+            # and at the very end, add the file to run
+            command_line_args += [full_file_path]
+
+            # run E+
+            if by_api:
+                # if by API then find the API wrapper relative to this workflow, import it, set up a callback, and run
+                eplus_dir = Path(__file__).parent.parent.absolute()
+                sys.path.insert(0, str(eplus_dir))
+                from pyenergyplus.api import EnergyPlusAPI
+                x = lambda msg: self.callback("API Callback: " + str(msg))
+                api = EnergyPlusAPI()
+                state = api.state_manager.new_state()
+                api.runtime.callback_message(state, x)
+                api.runtime.set_console_output_status(state, False)
```

Comment:
I had to comment out this line too. I think you renamed or removed this API call.

Reply:
That's weird...it should definitely be there.  It's in develop even: https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/api/runtime.py#L194.  I wonder how it is getting confused.

Reply:
This is no longer an issue per the PR comment from @JasonGlazer. :+1: